runtime/pretty/print.go

```diff
@@ -293,19 +287,35 @@ func (p ErrorPrettyPrinter) writeCodeExcerpts(
 			p.writeString(emptyLineNumbers)
 			p.writeString("\n")
 
-			// line number
-			p.writeString(lineNumberString)
+			var line string
+			for lineNumber := excerpt.startPos.Line - 1; lineNumber < excerpt.endPos.Line; lineNumber++ {
+				plainLineNumberString := strconv.Itoa(lineNumber + 1)
 
-			// code line
-			line := lines[excerpt.startPos.Line-1]
-			if len(line) > maxLineLength {
-				p.writeString(line[:maxLineLength])
-				p.writeString(excerptDots)
-			} else {
-				p.writeString(line)
-			}
+				// if the line number increases in digit length during the error,
+				// fill the extra space with blank spaces
+				if i = 0; i < lineNumberLength-len(plainLineNumberString) {
```

Comment:
`i` is the index of the excerpts, see https://github.com/onflow/cadence/pull/2197/files#diff-2c07386513ebe19503a3f5f086fbd480960c3d2a441c68f2a2fc29ec46008a63R254

Maybe rename that index to something more descriptive, like `excerptIndex` and then introduce a separate index variable here, instead of assigning/overwriting.